### ISSUE TITLE
qutebrowser hangs at start with my basedir

### ISSUE THREAD

**Issue by l29ah** [qutebrowser/qutebrowser#7801]
**Version info**:
qutebrowser v2.5.4
Git commit: 7724f7295-dirty on HEAD (2023-07-24 23:04:51 +1200)
Backend: QtWebEngine 6.5.2, based on Chromium 108.0.5359.220 (from api)
Qt: 6.5.2 (compiled 6.5.1)

CPython: 3.11.4
PyQt: 6.5.1

Qt wrapper: PyQt6 (via override)

**Does the bug happen if you start with `--temp-basedir`?**:
No.

**Description**
It hangs at start w/o even showing any windows. Consumes one CPU thread completely.
qutebrowser just hanged today, i killed it, and can't run it anymore. May have something to do with qt 6.5.1 -> 6.5.2 upgrade that happened in the meantime.

**How to reproduce**
Run qutebrowser.

**Comment by The-Compiler** [comment]
What happens if you run it in a terminal and then send it a SIGUSR1 while it's hanging?

**Comment by l29ah** [comment]
On Mon, Jul 24, 2023 at 11:19:43AM -0700, Florian Bruhin wrote:
> What happens if you run it in a terminal and then send it a SIGUSR1 while it's hanging?

Current thread 0x00007f93ed88d740 (most recent call first):
  File "/usr/lib/python3.11/site-packages/qutebrowser/mainwindow/tabwidget.py", line 607 in _text_to_width
  File "/usr/lib/python3.11/site-packages/qutebrowser/mainwindow/tabwidget.py", line 609 in _minimum_tab_size_hint_helper_uncached
  File "/usr/lib/python3.11/site-packages/qutebrowser/mainwindow/tabwidget.py", line 590 in minimumTabSizeHint
  File "/usr/lib/python3.11/site-packages/qutebrowser/mainwindow/tabwidget.py", line 305 in insertTab
  File "/usr/lib/python3.11/site-packages/qutebrowser/mainwindow/tabbedbrowser.py", line 650 in tabopen
  File "/usr/lib/python3.11/site-packages/qutebrowser/misc/sessions.py", line 459 in _load_window
  File "/usr/lib/python3.11/site-packages/qutebrowser/misc/sessions.py", line 496 in load
  File "/usr/lib/python3.11/site-packages/qutebrowser/misc/sessions.py", line 650 in load_default
  File "/usr/lib/python3.11/site-packages/qutebrowser/app.py", line 186 in _process_args
  File "/usr/lib/python3.11/site-packages/qutebrowser/app.py", line 155 in init
  File "/usr/lib/python3.11/site-packages/qutebrowser/app.py", line 108 in run
  File "/usr/lib/python3.11/site-packages/qutebrowser/qutebrowser.py", line 231 in main
  File "/usr/bin/qutebrowser", line 8 in <module>


-- 
()  ascii ribbon campaign - against html mail
/\  http://arc.pasp.de/   - against proprietary attachments

**Comment by The-Compiler** [comment]
...so it hangs when we ask Qt for the width of a text. Weird!

What happens if you do it multiple times, do you always get the same stacktrace? Does it work if you take a backup of your `~/.local/share/qutebrowser/sessions` and then start with `-R` (which ignores your session)?

**Comment by l29ah** [comment]
My guess is it's another piece of fallout from https://github.com/qutebrowser/qutebrowser/issues/5108

**Comment by l29ah** [comment]
Yes, the stacktrace is the same every time. qutebrowser starts fine with `-R`.

**Comment by l29ah** [comment]
A small `default.yml` reproducer for you: https://0x0.st/HegP.yml

**Comment by The-Compiler** [comment]
Can you reproduce this with something like `qutebrowser --temp-basedir ":session-load HegP.yml"`?

I don't see any qutebrowser hang with that. What I did see is my window manager really not liking the giant window title and things hanging, and I added a fix for that in 89b91105022be0216d8fb323b20fd15e1391af28 (I do recognize that we still need a proper fix for the error page URLs, but this might still be a problem with arbitrary `data:` URLs, so both are needed).

**Comment by The-Compiler** [comment]
ah, my bad I think, you might have to specify a full path (e.g. `qutebrowser --temp-basedir ":session-load ~/HegP.yml"`

**Comment by l29ah** [comment]
Tested with the current master, it results in no qutebrowser window being spawned and the process consuming 100% CPU for ~20s, then the window is spawned but doesn't react to anything while keeping the 100% CPU consumption. SIGINT's from console are being ignored also.

**Comment by The-Compiler** [comment]
With current master, I'm assuming you mean the `main` branch? master doesn't exist anymore. Either way I can still not reproduce it, but probably it will be gone once #3918 is fixed.
